src/repos/NotificationRepo.ts

```diff
@@ -8,28 +8,51 @@ import ArticleRepo from './ArticleRepo';
 import { Magazine } from '../entities/Magazine';
 import MagazineRepo from './MagazineRepo';
 
-const sendNewArticleNotification = async (
+/**
+ * General purpose function for sending notifications.
+ * @param user User object
+ * @param notifTitle Notification title
+ * @param notifBody Notificiation body
+ * @param uniqueData Any unique metadata to be sent in notif beyond uuid and notif type
+ * @param notifType Notification type (ex. 'new_article')
+ */
+const sendNotif = async (
   user: User,
-  article: Article,
-  publication: Publication,
-): Promise<void> => {
+  notifTitle: string,
+  notifBody: string,
+  uniqueData: Record<string, string>,
+  notifType: string,
+) => {
   const { deviceToken } = user;
-  const notifTitle = publication.name;
-  const notifBody = article.title;
 
-  const message = {
-    notification: {
-      title: notifTitle,
-      body: notifBody,
-    },
-    data: {
-      userID: user.uuid,
-      articleID: article.id,
-      articleURL: article.articleURL,
-      notificationType: 'new_article',
-    },
+  const notifData = {
+    title: notifTitle,
+    body: notifBody,
   };
 
+  const metaData = {
+    userID: user.uuid,
+    ...uniqueData,
+    notificationType: notifType,
+  };
+
+  let message = {};
+
+  // Determine notif format based on device type
+  if (user.deviceType === 'IOS') {
+    message = {
+      notification: notifData,
+      data: metaData,
+    };
+  } else {
+    message = {
+      data: {
```

Comment:
Android should migrate to iOS's format later so that we can have consistent responses. @ckdesir

Reply:
Don't think that will be possible.